pulser-core/pulser/backend/results.py

```diff
@@ -51,19 +53,19 @@ def _store(
             time: The relative time at which the observable was taken.
             value: The value of the observable.
         """
-        self._results.setdefault(observable.uuid, {})
-
-        if time in self._results[observable.uuid]:
+        _times = self._times.setdefault(observable.uuid, [])
+        if time in _times:
             raise RuntimeError(
                 f"A value is already stored for observable '{observable.tag}'"
                 f" at time {time}."
             )
         self._tagmap[observable.tag] = observable.uuid
-        self._results[observable.uuid][time] = value
+        _times.append(time)
+        self._results.setdefault(observable.uuid, []).append(value)
```

Comment:
```
_results = self._results.setdefault(observable.uuid, [])
assert len(_times) == len(_results)
_results.append(value)
```
is that also an idea?

Reply:
This should always be true we can `assert` it, yes

Reply:
Euh if there is no assertion then the append is not needed, right ? 
```suggestion
        self._results.setdefault(observable.uuid, [value])
```